### DISCUSSION TITLE
RFC: Establish well-known spec to communicate logs/cicd-stuff back to forges on request

### DISCUSSION THREAD

**Discussion by 6543** [woodpecker-ci/woodpecker#2159] [Request a Feature]
`< add long desc>`

TODOs:
- [ ] add it to https://en.wikipedia.org/wiki/Well-known_URI
- [ ] add schema
- [ ] define vocabulary
- [ ] add dedicated site for this into our docu

refs:
- https://www.rfc-editor.org/rfc/rfc5785

**Comment by 6543** [comment]
User Storry: I like to see the log of running pipelines of a ci at gitea

**Comment by oliverpool** [comment]
_Disclaimer: I am very novice regarding woodpecker and the whole CI setup. I hope that this comment is somehow relevant_

Instead of documenting such parameters under the well-known URLs, couldn't those parameters be passed around on each pipeline?

For instance:
- A commit was pushed to gitea, it generates a payload containing the URLs to report the pipelines status on the commit and sends this to Woodpecker (along the original payload)
```json
{
  // webhook payload comes here
  "report_endpoint": "https://gitea.example.com/api/commintinfo/<sha1>/..."
}
```
- when woodpecker receives this message, it can inform gitea of the discovered pipelines (or of any error), by POSTing to the `report_endpoint`
```json
{
  "pipelines:" [
    { "id": "...", "name": "...", "logs_endpoint": "https://woodpecker.example.com/api/..."}
  ],
  "run": 1 // in case of multiple runs on a given commit
}
```
- woodpecker could regularly PUT at this endpoint, to give updates regarding the pipeline status.

**Comment by 6543** [comment]
for vocabulary we should have a look at https://github.com/cdfoundation/sig-interoperability/blob/main/docs/tools-terminology.md#cicd-tools-and-technologies

**Comment by 6543** [comment]
how do we communicate, code-line-fiel specifig msgs:
![image](https://github.com/woodpecker-ci/woodpecker/assets/24977596/1a45f7a5-ad0d-4fb8-b608-ce40ed8ad614)

**Comment by 6543** [comment]
note: how do we signal to forge that there is such feature in the first place
idea: webhook response header contain a defined key
